# Calculate format=diff
diff --git a/modules.d/05busybox/module-setup.sh b/modules.d/05busybox/module-setup.sh
index 228ff8c..822a481 100755
--- a/modules.d/05busybox/module-setup.sh
+++ b/modules.d/05busybox/module-setup.sh
@@ -17,20 +17,18 @@ depends() {
 # called by dracut
 install() {
     local _i _progs _path _busybox
+    local NOT_FROM_BUSYBOX="${NOT_FROM_BUSYBOX} busybox losetup switch_root sh ,"
+    NOT_FROM_BUSYBOX="$(for _i in ${NOT_FROM_BUSYBOX}; do _i=${_i##*/}; [[ -n "$_i" ]] && echo -n " s/$_i//g;"; done)"
     _busybox=$(type -P busybox)
     inst $_busybox /usr/bin/busybox
-    for _i in $($_busybox | sed -ne '1,/Currently/!{s/,//g; s/busybox//g; p}')
+    for _i in $($_busybox | sed -ne '1,/Currently/!{'"${NOT_FROM_BUSYBOX}"' p}')
     do
         _progs="$_progs $_i"
     done
 
-    # FIXME: switch_root should be in the above list, but busybox version hangs
-    # (using busybox-1.15.1-7.fc14.i686 at the time of writing)
-
     for _i in $_progs; do
         _path=$(find_binary "$_i")
         [ -z "$_path" ] && continue
         ln_r /usr/bin/busybox $_path
     done
 }
-
