# Calculate format=diff

diff -ru a/calculate-lib-3.3.0.3/pym/calculate/lib/variables/linux.py b/calculate-lib-3.3.0.3/pym/calculate/lib/variables/linux.py
--- a/calculate-lib-3.3.0.3/pym/calculate/lib/variables/linux.py
+++ b/calculate-lib-3.3.0.3/pym/calculate/lib/variables/linux.py
@@ -32,11 +32,13 @@
                      "CLS":"Calculate Linux Scratch",
                      "CSS":"Calculate Scratch Server",
                      "CMC":"Calculate Media Center",
+                     "CLDM":"Calculate Linux Desktop",
                      "Gentoo":"Gentoo"}
 
-    dictLinuxSubName = {"CLD":"KDE", "CLDX":"XFCE", "CLDG":"GNOME"}
+    dictLinuxSubName = {"CLD":"KDE", "CLDX":"XFCE", "CLDG":"GNOME", "CLDM":"Mate", }
 
     dictNameSystem = {'CDS':'server',
+                      'CLDM':'desktop',
                       'CLD':'desktop',
                       'CLDG':'desktop',
                       'CLDX':'desktop',
--- a/calculate-lib-3.3.0.3/pym/calculate/lib/utils/device.py
+++ b/calculate-lib-3.3.0.3/pym/calculate/lib/utils/device.py
@@ -340,6 +340,8 @@
                 dictInfo['name'] = "CLD"
             elif "GNOME" in issue:
                 dictInfo['name'] = "CLDG"
+            elif "Mate" in issue:
+                dictInfo['name'] = "CLDM"
         elif "Scratch" in issue:
             dictInfo['name'] = "CLS"
     else:
